Dockerfile.golang.base

```diff
@@ -3,8 +3,10 @@ ARG PRESIDIO_DEPS_LABEL=latest
 
 FROM ${REGISTRY}/presidio-golang-deps:${PRESIDIO_DEPS_LABEL}
 
+
+COPY Gopkg.lock Gopkg.toml $GOPATH/src/github.com/Microsoft/presidio/
 WORKDIR $GOPATH/src/github.com/Microsoft/presidio
-ADD . $GOPATH/src/github.com/Microsoft/presidio
+RUN dep ensure -vendor-only
 
-RUN dep ensure
+ADD . $GOPATH/src/github.com/Microsoft/presidio
```

Comment:
If I understand correctly.
you divided it just so the layers will be easier for caching?
line 11, is now doing what the old line 7 was doing. right?

so functionality wise there is no difference. right?

Reply:
yes. functionality wise the "docker build" has the same results.
this way, if you change your code (which happens more frequently than changing gopkg), dep ensure will not run which is the major improvement here.